tests/e2e: more debug info on dumps

willfindlay opened the pull request:
This series adds a bunch more debug info to e2e test dumps. See commits.

kevsecurity commented on tests/e2e/helpers/dumpinfo.go:
```diff
@@ -218,6 +221,27 @@ func dumpMetrics(port string, podName string, exportDir string) {
+// back on.
+func StartMetricsDumper(ctx context.Context, exportDir string, interval time.Duration) {
+	go func() {
+		for {
```
Maybe use a ticker, or the *amazing* timer package here?

kkourt commented on tests/e2e/helpers/dumpinfo.go:
```diff
@@ -218,6 +221,28 @@ func dumpMetrics(port string, podName string, exportDir string) {
+					}
+				}
+			case <-ctx.Done():
+				return
```
Does it make sense to dump stats at ctx.Done() as well while we are at it?

kkourt commented on tests/e2e/helpers/dumpinfo.go:
```diff
@@ -218,6 +221,28 @@ func dumpMetrics(port string, podName string, exportDir string) {
+// back on.
+func StartMetricsDumper(ctx context.Context, exportDir string, interval time.Duration) {
+	go func() {
+		timer := time.NewTicker(interval)
```
nit: rename to ticker?

kkourt reviewed (approved):
LGTM, just some minor comments

willfindlay commented:
@kkourt I addressed your comments. Also realized at the same time that we were never actually canceling our context in the tests so I had to refactor the runner a bit (it's in the new first commit).
